### ISSUE TITLE
Provide custom hard tab behavior for code blocks according to language (e.g., go / golang)

### ISSUE THREAD

**Issue by TJM** [DavidAnson/markdownlint#383]
Related to: #282 

golang uses "hard tabs" (much to my dismay), and that causes the linter to display a huge amount of violations. The `no-hard-tabs` rule can be turned off, either globally, or in code blocks, but I assert that it should *default* to off in a code block labeled as "golang" ... like:

```
```go
```

or

```
```golang
```

I am not sure what the level of effort required to get the "type" of a code block would be, then to set a different default for golang (and any other language that insists on using tabs), but I think it would make the tool much more user friendly.

Thanks,
Tommy
